internal/core/crdt/collection.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package crdt
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/defradb/datastore"
+	"github.com/sourcenetwork/defradb/internal/core"
+	"github.com/sourcenetwork/defradb/internal/keys"
+)
+
+// Collection is a simple CRDT type that tracks changes to the contents of a
+// collection in a similar way to a document composite commit, only simpler,
+// without the need to track status and a simpler [Merge] function.
+type Collection struct {
+	store datastore.DSReaderWriter
+
+	// schemaVersionKey is the schema version datastore key at the time of commit.
+	//
+	// It can be used to identify the collection datastructure state at the time of commit.
+	schemaVersionKey keys.CollectionSchemaVersionKey
+}
+
+var _ core.ReplicatedData = (*Collection)(nil)
+
+func NewCollection(store datastore.DSReaderWriter, schemaVersionKey keys.CollectionSchemaVersionKey) *Collection {
+	return &Collection{
+		store:            store,
+		schemaVersionKey: schemaVersionKey,
+	}
+}
+
+func (c *Collection) Merge(ctx context.Context, other core.Delta) error {
+	// Collection merges don't actually need to do anything, as the delta is empty,
+	// and doc-level merges are handled by the document commits.
+	return nil
+}
+
+func (c *Collection) Append() *CollectionDelta {
```

Comment:
suggestion: `Set` might be a better name for this. There is no reason for it to be different than the composite CRDT.

Reply:
I renamed composite.Set to `Append`

Reply:
Append is equally a bad name for something that doesn't append anything though. Set does set fields in the delta object.

Reply:
A new delta object, is sets nothing on the host.  Although `Append` doesnt append to the host really either.  Both names are bad, but they are now consistent at least.

The two functions are constructors, nothing more.  They could be renamed `NewDelta` perhaps.

Note: I don't see this as something terribly important, and am not very keen on thinking much about it.

Reply:
Can you leave it as `Set` for now. It doesn't make sense to me to chose the objectively less accurate name `Append` when `Set` was there previously. I have a vague memory of John deciding to use `Set` because it would have been the correct naming in CRDT lingo. If that's not the case and you really don't want to use `Set`, `NewDelta` is definitely the better choice.

Reply:
done :)